File: medicare2/models/patient.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class Report(BaseModel):
    name: str
    type: str
    date: str
    file_id: Optional[str] = None
    fileType: str = "pdf"  # Default to PDF
    description: Optional[str] = None

class PatientRecord(BaseModel):
    visitDate: str
    visitTime: str
    condition: str
    doctor: str
    prescription: Optional[str] = None
    status: str

class PatientCreate(BaseModel):
    patientName: str
    condition: str
    doctor: str
    lastVisit: str
    lastVisitTime: str
    status: str
    prescription: Optional[str] = None

class Patient(PatientCreate):
    id: str
    reports: List[Report] = []
    previousRecords: List[PatientRecord] = []

class PatientUpdate(BaseModel):
    patientName: Optional[str] = None
    condition: Optional[str] = None
    doctor: Optional[str] = None
    lastVisit: Optional[str] = None
    lastVisitTime: Optional[str] = None
    status: Optional[str] = None
    prescription: Optional[str] = None

File: medicare1/src/app/emr/models/Patient.ts
import { Report } from './Report';
import { PatientRecordWithUser } from './PatientRecordWithUser';

export interface Patient {
  id: string;
  patientName: string;
  condition: string;
  doctor: string;
  visitDate: string;
  visitTime: string;
  status: string;
  prescription: string;
  reports: Report[];
  previousRecords: PatientRecordWithUser[];
}
